lib/ansiblelint/__main__.py

```diff
@@ -20,23 +20,98 @@
+import tempfile
 from ansiblelint import cli, default_rulesdir
-from ansiblelint.runner import Runner
-from ansiblelint.rules import RulesCollection
-from ansiblelint.utils import get_playbooks_and_roles
-from ansiblelint.utils import normpath, initialize_logger
 from ansiblelint.generate_docs import rules_as_rst
 from typing import Any, Set
 
 
 _logger = logging.getLogger(__name__)
 
+MODULE_STUB = """
+# This is a fake module used to make ansible(-lint) happy
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    return AnsibleModule(
+        argument_spec=dict(
+            data=dict(default=None),
+            path=dict(default=None, type=str),
+            file=dict(default=None, type=str),
+        )
+    )
+"""
+
+
+def initialize_logger(level: int = 0) -> None:
+    """Set up the global logging level based on the verbosity number."""
+    VERBOSITY_MAP = {
+        0: logging.NOTSET,
+        1: logging.INFO,
+        2: logging.DEBUG
+    }
+
+    handler = logging.StreamHandler()
+    formatter = logging.Formatter('%(levelname)-8s %(message)s')
+    handler.setFormatter(formatter)
+    logger = logging.getLogger(__package__)
+    logger.addHandler(handler)
+    # Unknown logging level is treated as DEBUG
+    logging_level = VERBOSITY_MAP.get(level, logging.DEBUG)
+    logger.setLevel(logging_level)
+    # Use module-level _logger instance to validate it
+    _logger.debug("Logging initialized to level %s", logging_level)
+
+
+@contextmanager
+def bootstrap_ansible(options):
+    """Perform Ansible boostraping steps to add custom module paths.
+
+    * Alter ANSIBLE_LIBRARY to inject custom paths before ansible modules are
+    loaded. See https://github.com/ansible/ansible/issues/69758
+    """
+    BootstrapContext = collections.namedtuple(
+        'BootstrapContext',
+        'change_module_path tmp_dir')
+    changed_module_path = False
+    tmp_dir = None
+
+    lib_paths = os.environ.get(
+        'ANSIBLE_LIBRARY',
+        "~/.ansible/plugins/modules:/usr/share/ansible/plugins/modules").split(":")
+    # add plugins/modules to module paths if it exists
+    p = pathlib.Path.cwd() / "plugins" / "modules"
+    if p.exists() and str(p) not in lib_paths:
+        _logger.info(f"Adding {str(p)} to module path")
+        lib_paths.append(str(p))
+        changed_module_path = True
+
+    if options.stub_modules:
+        # creates temporary directory for stub modules (auto-cleaned)
+        tmp_dir = tempfile.TemporaryDirectory()
+        lib_paths.append(tmp_dir.name)
+        _logger.info(f"Adding {tmp_dir.name} to module path")
+        changed_module_path = True
+
+        for m in options.stub_modules:
+            with (pathlib.Path(tmp_dir.name) / f"{m}.py") as f:
+                _logger.debug(f"Generating fake module stub: {f.name}")
+                f.write_text(MODULE_STUB)
+
+    if changed_module_path:
+        os.environ['ANSIBLE_LIBRARY'] = ":".join(lib_paths)
```

Comment:
This is unnecessary. The only thing needed is
```suggestion
        from ansible.plugins.loader import module_loader
        module_loader.add_directory(tmp_dir.name)
```

In fact, it's already used in other places: https://github.com/ansible/ansible-lint/blob/0cf285b/lib/ansiblelint/utils.py#L292-L294.